### PR TITLE
Support node preservation for accordion and tabset

### PR DESCRIPTION
**Pull request by jiayihu**
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [ ] added/updated any applicable demos.

This resolves https://github.com/ng-bootstrap/ng-bootstrap/issues/1161 and https://github.com/ng-bootstrap/ng-bootstrap/issues/1167 with an optional Input `preserveNodes` which uses `hidden` property if `true` for content hiding.

### CODE REVIEW

**Comment by jiayihu** [comment]
Apparently my new tests are failing on Travis while being okay locally. Also I can't get the logs because Travis says there's an error, so I'll look at it later.

Meanwhile any feedback is very appreciated.

**Comment by jiayihu** [comment]
I think Travis build failure on IE10 is because `hidden` property is not supported, but I would anyway expect the property to be added by Angular. Is it my fault?

**Comment by jiayihu** [comment]
I solved using `hidden` attribute. Bootstrap 4 `reboot` will take care of the rest adding a [hidden polyfill](https://github.com/twbs/bootstrap/blob/v4-dev/scss/_reboot.scss#L482).

File: src/tabset/tabset.ts [hunk 1]
```diff
@@ -90,7 +90,10 @@ export interface NgbTabChangeEvent {
     </ul>
     <div class="tab-content">
       <template ngFor let-tab [ngForOf]="tabs">
-        <div class="tab-pane active" *ngIf="tab.id === activeId" role="tabpanel"
+        <div class="tab-pane active"
+          *ngIf="preserveNodes || tab.id === activeId"
+          [attr.hidden]="isHidden(tab.id)"
```
Comment (pkozlowski-opensource) [inline src/tabset/tabset.ts]: You should _not_ be binding to the `hidden` attribute but rather use Bootstrap's `active` / `show` CSS classes. Check the markup generated by Bootstrap 4 and follow it: https://v4-alpha.getbootstrap.com/components/navs/#javascript-behavior

File: src/tabset/tabset.ts [hunk 2]
```diff
@@ -113,6 +116,11 @@ export class NgbTabset implements AfterContentChecked {
   @Input() justify: 'start' | 'center' | 'end';
 
   /**
+   * Whether the closed tabs should be hidden without destroying them
+   */
+  @Input() preserveNodes: boolean;
```
Comment (pkozlowski-opensource) [inline src/tabset/tabset.ts]: I would rather name it 'destroyOnHide' and give it `true` value by default.

**Comment by pkozlowski-opensource** [comment]
@jiayihu thnx for the PR. I've left a couple of comments in the code. We would like to merge it but before it happens we need the following:
* please address the comments in the code
* please split your PR into 2 separate ones: one PR for tabset and another one for accordion

Thnx!

**Comment by jiayihu** [comment]
@pkozlowski-opensource Thanks for your feedback. I'll close this PR in favor of #1369 and #1370

**Pull request closed without merging**